pkg/compare/compare.go

```diff
@@ -227,24 +226,27 @@ func (o *Options) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string
 	var fs fs.FS
 	o.builder = f.NewBuilder()
 
-	if o.templatesDir == "" {
-		return kcmdutil.UsageErrorf(cmd, noRefDirectoryWasPassed)
+	if o.referenceConfig == "" {
+		return kcmdutil.UsageErrorf(cmd, noRefFileWasPassed)
 	}
-	if _, err := os.Stat(o.templatesDir); os.IsNotExist(err) && !isURL(o.templatesDir) {
-		return fmt.Errorf(refDirNotExistsError)
+	if _, err := os.Stat(o.referenceConfig); os.IsNotExist(err) && !isURL(o.referenceConfig) {
+		return fmt.Errorf(refFileNotExistsError)
 	}
 
-	if isURL(o.templatesDir) {
-		fs = HTTPFS{baseURL: o.templatesDir, httpGet: httpgetImpl}
+	referenceDir := filepath.Dir(o.referenceConfig)
+	referenceFileName := filepath.Base(o.referenceConfig)
+	if isURL(o.referenceConfig) {
+		referenceDir = strings.Replace(referenceDir, "/", "//", 1)
```

Comment:
Is this becuase `filepath.Base` will turn `http://` to `http:/`?

Reply:
Yes, I think it's still nicer than URL parse

Reply:
Could you add a comment to explain why its there. One of those odd things that I find comments help save people a lot of time :)

Reply:
Done